model_compression_toolkit/target_platform_capabilities/schema/v1.py

```diff
@@ -270,69 +314,97 @@ def clone_and_edit(self, **kwargs) -> 'QuantizationConfigOptions':
             **kwargs: Keyword arguments to edit in each configuration.
 
         Returns:
-            A new instance of QuantizationConfigOptions with updated configurations.
+            QuantizationConfigOptions: A new instance with updated configurations.
         """
-        updated_base_config = replace(self.base_config, **kwargs)
-        updated_configs = [
-            replace(cfg, **kwargs) for cfg in self.quantization_configurations
-        ]
-        return replace(self, base_config=updated_base_config, quantization_configurations=tuple(updated_configs))
+        # Clone and update base_config
+        updated_base_config = self.base_config.clone_and_edit(**kwargs) if self.base_config else None
+
+        # Clone and update all configurations
+        updated_configs = tuple(cfg.clone_and_edit(**kwargs) for cfg in self.quantization_configurations)
 
-    def clone_and_edit_weight_attribute(self, attrs: List[str] = None, **kwargs) -> 'QuantizationConfigOptions':
+        return self.model_copy(update={
+            'base_config': updated_base_config,
+            'quantization_configurations': updated_configs
+        })
+
+    def clone_and_edit_weight_attribute(self, attrs: Optional[List[str]] = None, **kwargs) -> 'QuantizationConfigOptions':
```

Comment:
I don't remember if we discussed this, but this function and `clone_and_map_weights_attr_keys` don't seem like something that should be part of the schema, but rather an external functionality (although it is probably not possible because of the immutability? is that the problem?)